# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/interception.h
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/interception.h
--- a/security/sandbox/chromium/sandbox/win/src/interception.h	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/interception.h	Wed Jun 12 11:10:48 2019 +0100
@@ -156,7 +156,7 @@
   // value is the current size of the buffer, and alignment is specified in
   // bytes.
   static inline size_t RoundUpToMultiple(size_t value, size_t alignment) {
-    return ((value + alignment -1) / alignment) * alignment;
+    return ((value + alignment - 1) / alignment) * alignment;
   }
 
   // Sets up a given buffer with all the information that has to be transfered
@@ -176,14 +176,16 @@
   // (the actual interception info), and the current size of the buffer will
   // decrease to account the space used by this method.
   bool SetupDllInfo(const InterceptionData& data,
-                    void** buffer, size_t* buffer_bytes) const;
+                    void** buffer,
+                    size_t* buffer_bytes) const;
 
   // Fills up the part of the transfer buffer that corresponds to a single
   // function to patch.
   // dll_info points to the dll being updated with the interception stored on
   // data. The buffer pointer and remaining size are updated by this call.
   // Returns true on success.
-  bool SetupInterceptionInfo(const InterceptionData& data, void** buffer,
+  bool SetupInterceptionInfo(const InterceptionData& data,
+                             void** buffer,
                              size_t* buffer_bytes,
                              DllPatchInfo* dll_info) const;
 
@@ -194,7 +196,8 @@
   // Allocates a buffer on the child's address space (returned on
   // remote_buffer), and fills it with the contents of a local buffer.
   // Returns SBOX_ALL_OK on success.
-  ResultCode CopyDataToChild(const void* local_buffer, size_t buffer_bytes,
+  ResultCode CopyDataToChild(const void* local_buffer,
+                             size_t buffer_bytes,
                              void** remote_buffer) const;
 
   // Performs the cold patch (from the parent) of ntdll.
@@ -235,30 +238,29 @@
 // (WINAPI = with the "C" underscore).
 #if SANDBOX_EXPORTS
 #if defined(_WIN64)
-#define MAKE_SERVICE_NAME(service, params) "Target" # service "64"
+#define MAKE_SERVICE_NAME(service, params) "Target" #service "64"
 #else
-#define MAKE_SERVICE_NAME(service, params) "_Target" # service "@" # params
+#define MAKE_SERVICE_NAME(service, params) "_Target" #service "@" #params
 #endif
 
-#define ADD_NT_INTERCEPTION(service, id, num_params) \
-  AddToPatchedFunctions(kNtdllName, #service, \
+#define ADD_NT_INTERCEPTION(service, id, num_params)        \
+  AddToPatchedFunctions(kNtdllName, #service,               \
                         sandbox::INTERCEPTION_SERVICE_CALL, \
                         MAKE_SERVICE_NAME(service, num_params), id)
 
-#define INTERCEPT_NT(manager, service, id, num_params) \
-  ((&Target##service) ? \
-    manager->ADD_NT_INTERCEPTION(service, id, num_params) : false)
+#define INTERCEPT_NT(manager, service, id, num_params)                        \
+  ((&Target##service) ? manager->ADD_NT_INTERCEPTION(service, id, num_params) \
+                      : false)
 
 // When intercepting the EAT it is important that the patched version of the
 // function not call any functions imported from system libraries unless
 // |TargetServices::InitCalled()| returns true, because it is only then that
 // we are guaranteed that our IAT has been initialized.
-#define INTERCEPT_EAT(manager, dll, function, id, num_params) \
-  ((&Target##function) ? \
-    manager->AddToPatchedFunctions(dll, #function, sandbox::INTERCEPTION_EAT, \
-                                   MAKE_SERVICE_NAME(function, num_params), \
-                                   id) : \
-    false)
+#define INTERCEPT_EAT(manager, dll, function, id, num_params)             \
+  ((&Target##function) ? manager->AddToPatchedFunctions(                  \
+                             dll, #function, sandbox::INTERCEPTION_EAT,   \
+                             MAKE_SERVICE_NAME(function, num_params), id) \
+                       : false)
 #else  // SANDBOX_EXPORTS
 #if defined(_WIN64)
 #define MAKE_SERVICE_NAME(service) &Target##service##64
@@ -266,10 +268,10 @@
 #define MAKE_SERVICE_NAME(service) &Target##service
 #endif
 
-#define ADD_NT_INTERCEPTION(service, id, num_params) \
-  AddToPatchedFunctions(kNtdllName, #service, \
-                        sandbox::INTERCEPTION_SERVICE_CALL, \
-                        (void*)MAKE_SERVICE_NAME(service), id)
+#define ADD_NT_INTERCEPTION(service, id, num_params)            \
+  AddToPatchedFunctions(                                        \
+      kNtdllName, #service, sandbox::INTERCEPTION_SERVICE_CALL, \
+      reinterpret_cast<void*>(MAKE_SERVICE_NAME(service)), id)
 
 #define INTERCEPT_NT(manager, service, id, num_params) \
   manager->ADD_NT_INTERCEPTION(service, id, num_params)
@@ -279,8 +281,9 @@
 // |TargetServices::InitCalled()| returns true, because it is only then that
 // we are guaranteed that our IAT has been initialized.
 #define INTERCEPT_EAT(manager, dll, function, id, num_params) \
-  manager->AddToPatchedFunctions(dll, #function, sandbox::INTERCEPTION_EAT, \
-                                 (void*)MAKE_SERVICE_NAME(function), id)
+  manager->AddToPatchedFunctions(                             \
+      dll, #function, sandbox::INTERCEPTION_EAT,              \
+      reinterpret_cast<void*>(MAKE_SERVICE_NAME(function)), id)
 #endif  // SANDBOX_EXPORTS
 
 }  // namespace sandbox